openedx/core/djangoapps/content_libraries/collections/rest_api/v1/views.py

```diff
@@ -0,0 +1,110 @@
+"""
+Collections API Views
+"""
+
+from __future__ import annotations
+
+from django.http import Http404
+
+# from rest_framework.generics import GenericAPIView
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.status import HTTP_405_METHOD_NOT_ALLOWED
+
+from opaque_keys.edx.locator import LibraryLocatorV2
+
+from openedx.core.djangoapps.content_libraries import api, permissions
+from openedx.core.djangoapps.content_libraries.serializers import (
+    ContentLibraryCollectionSerializer,
+    ContentLibraryCollectionCreateOrUpdateSerializer,
+)
+
+from openedx_learning.api.authoring_models import Collection
+from openedx_learning.api import authoring as authoring_api
+
+
+class LibraryCollectionsView(ModelViewSet):
+    """
+    Views to get, create and update Library Collections.
+    """
+
+    serializer_class = ContentLibraryCollectionSerializer
+
+    def retrieve(self, request, lib_key_str, pk=None):
+        """
+        Retrieve the Content Library Collection
+        """
+        try:
+            collection = authoring_api.get_collection(pk)
+        except Collection.DoesNotExist as exc:
+            raise Http404 from exc
+
+        # Check if user has permissions to view this collection by checking if
+        # user has permission to view the Content Library it belongs to
+        library_key = LibraryLocatorV2.from_string(lib_key_str)
+        api.require_permission_for_library_key(library_key, request.user, permissions.CAN_VIEW_THIS_CONTENT_LIBRARY)
```

Comment:
@yusuf-musleh We need to make sure the collection is actually in the requested library.. what do you think of this approach?

```
# TODO: Update api.require_permission_for_library_key to return the library object so we're not refetching
library_obj = api.require_permission_for_library_key(library_key, request.user, permissions.CAN_EDIT_THIS_CONTENT_LIBRARY)
collection = authoring_api.get_learning_package_collections(library_obj.learning_package).filter(id=pk)
```